geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionEntry.java

```diff
@@ -1329,14 +1329,23 @@ public Object prepareValueForCache(RegionEntryContext r,
             data = event.getCachedSerializedNewValue();
           } else if (val instanceof CachedDeserializable) {
             data = ((CachedDeserializable) val).getSerializedValue();
+            if (event != null) {
```

Comment:
Instead of adding the new check in three places I would suggest that you add it to a single place like so:
          if (event != null && event.getCachedSerializedNewValue() != null) {
            data = event.getCachedSerializedNewValue();
          } else {
            if ((val instanceof CachedDeserializable) {
                 ....
             } else if (val instanceof PdxInstance) {
                 ....
             } else {
                ...
              }
             if (event != null) {
                event.setCachedSerializedNewValue(data);
              }
          }